client/app/widgets/Comments/__tests__/Comments.spec.jsx

```diff
@@ -83,41 +82,34 @@ const component = <Comments formProps={formProps} />;
 
 describe('Comments', () => {
   beforeEach(() => {
-    axiosPostSpy = jest.spyOn(axios, 'post').mockImplementation(() => Promise.resolve({
+    jest.spyOn(axios, 'post').mockImplementation(() => Promise.resolve({
       data: { comment },
     }));
-    axiosDeleteSpy = jest.spyOn(axios, 'delete').mockImplementation(() => Promise.resolve({
+    jest.spyOn(axios, 'delete').mockImplementation(() => Promise.resolve({
       data: { id },
     }));
   });
 
   it('renders correctly', () => {
-    let wrapper = null;
-    expect(() => {
-      wrapper = mount(component);
-    }).not.toThrow();
-    expect(wrapper).not.toBeNull();
+    render(component);
+    expect(() => component).not.toThrow();
+    expect(component).not.toBeNull();
   });
 
   it('add and delete a comment', async () => {
-    const wrapper = mount(component);
-    expect(wrapper.find('.comment').exists()).toEqual(false);
-    act(() => {
-      wrapper
-        .find('input[name="comment[comment]"]')
-        .simulate('change', { currentTarget: { value } });
+    render(component);
+    expect(screen.queryByRole('article')).toBeNull();
 
-      wrapper.find('select#comment_visibility').prop('onChange')({
-        currentTarget: { value: 'private' },
-      });
+    act(() => {
+      userEvent.type(screen.getByRole('textbox'));
+      userEvent.selectOptions(screen.getByRole('combobox'), 'private');
+      userEvent.click(screen.getByRole('button', { name: 'Submit' }));
     });
-    wrapper.find('input[type="submit"]').simulate('click');
-    await axiosPostSpy();
-    wrapper.update();
-    expect(wrapper.find('.comment').exists()).toEqual(true);
-    wrapper.find('.storyActionsDelete').find('a').simulate('click');
-    await axiosDeleteSpy();
-    wrapper.update();
-    expect(wrapper.find('.comment').exists()).toEqual(false);
+
+    await waitFor(() => expect(screen.queryByRole('article')).not.toBeNull());
```

Comment:
Would it also make sense to check how the comment is rendered, e.g. get the text displayed in the comment?

Reply:
That is more comprehensive. I will add it.